backend/src/routes/getCurrentExpense.ts

```diff
@@ -0,0 +1,48 @@
+import { and, eq, gte } from 'drizzle-orm/pg-core/expressions';
+import type { FastifyInstance } from 'fastify';
+
+import { usdcBalance, usdcTransfer } from '@/db/schema';
+
+const addressRegex = /^0x0[0-9a-fA-F]{63}$/;
+
+export function getCurrentExpenseRoute(fastify: FastifyInstance) {
+  fastify.get('/get_current_expense', async (request, reply) => {
+    const { address } = request.query as { address?: string };
+
+    if (!address) {
+      return reply.status(400).send({ error: 'Address is required.' });
+    }
+    // Validate address format
+    if (!addressRegex.test(address)) {
+      return reply.status(400).send({ error: 'Invalid address format.' });
+    }
+
+    try {
+      // Get the current date
+      const currentDate = new Date();
+
+      // Calculate the date 7 days ago
+      const sevenDaysAgo = new Date(currentDate);
+      sevenDaysAgo.setDate(currentDate.getDate() - 7);
+      // Use Drizzle ORM to find expense by address
+      const expenses = await fastify.db.query.usdcTransfer
+        .findMany({
+          where: and(
+            eq(usdcTransfer.fromAddress, address),
+            gte(usdcTransfer.createdAt, sevenDaysAgo),
+          ),
+        })
+        .execute();
+
+      let totalAmount = 0;
+      for (let i = 0; i < expenses.length; i++) {
+        totalAmount += Number(expenses[i].amount);
+      }
```

Comment:
```suggestion
     const expenses = await fastify.db.select({
      totalAmount: sql`sum(${usdcTransfer.amount})`.mapWith(Number)
     }).from(usdcTransfer).where(
      and(
        eq(usdcTransfer.fromAddress, address),
        between(usdcTransfer.createdAt, sql`now()`, sql`now() - interval '7 days'`)
      )
     )
     .execute();

     const totalAmount =  expenses[0].totalAmount ?? 0;
```

Double check the query, but it can be done fully in SQL.